tasks/marketplacesvm-push-disk-images/marketplacesvm-push-disk-images.yaml

```diff
@@ -140,4 +144,9 @@ spec:
         pushsource-ls "staged:${BASE_DIR}"
 
         # Process the push
-        marketplacesvm_push_wrapper --debug --source "${BASE_DIR}" --starmap-file "$STARMAP_MAPPING_FILE"
+        pre_push="$(params.prePush)"
+        add_args=""
+        if [[ "${pre_push,,}" == "true" ]]; then
+          add_args="--nochannel"
+        fi
+        marketplacesvm_push_wrapper --debug $add_args --source "${BASE_DIR}" --starmap-file "$STARMAP_MAPPING_FILE"
```

Comment:
I'm surprised shellcheck does not complain about `$add_args` :) So I guess it's fine. Otherwise you could use an array for this, e.g. 
```
EXTRA_ARGS=()
if prepush; then
  EXTRA_ARGS=(--nochannel)
marketplacesvm_push_wrapper ... "${EXTRA_ARGS[@]}"` ...
```

But ok to leave as is, up to you.